nupic/algorithms/anomaly.py

```diff
@@ -167,10 +178,35 @@ def __str__(self):
   def __setstate__(self, state):
     """deserialization"""
     self.__dict__.update(state)
-
     if not hasattr(self, '_mode'):
```

Comment:
Since the change allows `_mode` to have a callable as its value, what would be the implications on pickling and unpickling? Would it work properly in that case? Is there a test for it?

CC @scottpurdy